packages/roosterjs-editor-core/lib/coreApi/getPendableFormatState.ts

```diff
@@ -0,0 +1,84 @@
+import { contains, getTagOfNode, PendableFormatNames, Position } from 'roosterjs-editor-dom';
+import { EditorCore, NodePosition, NodeType, PendableFormatState } from 'roosterjs-editor-types';
+
+/**
+ * @param core The EditorCore object
+ * @param forceGetStateFromDOM If set to true, will force get the format state from DOM tree.
+ * @returns The cached format state if it exists. If the cached postion do not exist, search for pendable elements in the DOM tree and return the pendable format state.
+ */
+export function getPendableFormatState(
+    core: EditorCore,
+    forceGetStateFromDOM: boolean = false
+): PendableFormatState {
+    const range = core.api.getSelectionRange(core, true /** tryGetFromCache*/);
+    const isRangeCollapsed = range && range.collapsed;
+    const cachedPendableFormatState = core.pendingFormatState.pendableFormatState;
+    const cachedPosition = core.pendingFormatState.pendableFormatPosition?.normalize();
+    const currentPosition = isRangeCollapsed && Position.getStart(range).normalize();
+    const isSamePosition = currentPosition && currentPosition.equalTo(cachedPosition);
+
+    if (range && cachedPendableFormatState && isSamePosition && !forceGetStateFromDOM) {
+        return cachedPendableFormatState;
+    } else {
+        return queryCommandStateFromDOM(core, isRangeCollapsed, currentPosition);
+    }
+}
+
+const PendableStyleCheckers: Record<
+    PendableFormatNames,
+    (tagName: string, style: CSSStyleDeclaration) => boolean
+> = {
+    isBold: (tag, style) =>
+        tag == 'B' ||
+        tag == 'STRONG' ||
+        parseInt(style.fontWeight) >= 700 ||
+        ['bold', 'bolder'].includes(style.fontWeight),
+    isUnderline: (tag, style) => tag == 'U' || style.textDecoration.indexOf('underline') >= 0,
+    isItalic: (tag, style) => tag == 'I' || tag == 'EM' || style.fontStyle === 'italic',
+    isSubscript: (tag, style) => tag == 'SUB' || style.verticalAlign === 'sub',
+    isSuperscript: (tag, style) => tag == 'SUP' || style.verticalAlign === 'super',
+    isStrikeThrough: (tag, style) =>
+        tag == 'S' || tag == 'STRIKE' || style.textDecoration.indexOf('line-through') >= 0,
+};
+
+const NonPendableStyleCheckers: Record<
```

Comment:
We need a better name. How about "CssFalsyCheckers", and add some comment to explain why we need this